src/Sentry.Unity/UnityLogException.cs

```diff
@@ -149,6 +164,215 @@ public SentryException ToSentryException()
+            @"(\((?<module>.+?)\)\s*)?" + // (Unity), (Mono JIT Code), ...
+            @"(\((?<wrapper>.+?)\)\s*)?" + // (wrapper managed-to-native),
+            @"(\[(?<file>.+?):(?<line>\d+)\]\s*)?" + // [File.cs:123]
+            @"(?<function>.+)"); // StackWalker::GetCurrentCallstack, UnityEngine.Logger:Log (UnityEngine.LogType,UnityEngine.Object,string,object[]),
+
+        private SentryException ConvertFullStackTraceImpl()
+        {
+            var frames = new List<SentryStackFrame>();
+            var lines = LogStackTrace.Split('\n');
+            foreach (var traceLine in lines)
+            {
+                var item = traceLine.TrimEnd('\r');
+                if (string.IsNullOrWhiteSpace(item))
+                {
+                    continue;
+                }
+
+                var match = FullStackFramePattern.Match(item);
+                if (!match.Success)
+                {
+                    // we did something wrong, failed the check
+                    Debug.Log("failed parsing " + item);
+                    frames.Add(new SentryStackFrame
+                    {
+                        Function = item,
+                    });
+                }
+                else
+                {
+                    var module = match.Groups["module"].Value;
+                    var function = match.Groups["function"].Value;
+
+                    string? wrapper = null;
+                    if (match.Groups["wrapper"].Success)
+                    {
+                        wrapper = match.Groups["wrapper"].Value;
+                    }
+
+                    string? file = null;
+                    if (match.Groups["file"].Success)
+                    {
+                        file = match.Groups["file"].Value;
+                    }
+
+                    int? line = null;
+                    if (match.Groups["line"].Success)
+                    {
+                        line = Convert.ToInt32(match.Groups["line"].Value);
+                    }
+
+                    frames.Add(new SentryStackFrame
+                    {
+                        Module = module,
+                        Function = function,
+                        FileName = file,
+                        LineNumber = line,
+                        InApp = GuessInApp(module, function, wrapper, file),
+                    });
+                }
+            }
+
+            frames.Reverse();
+            var stacktrace = new SentryStackTrace();
+            foreach (var frame in frames)
+            {
+                stacktrace.Frames.Add(frame);
+            }
+
+            return new SentryException
+            {
+                Stacktrace = stacktrace,
+                Type = LogType.ToString(),
+                Value = LogString,
+                Mechanism = new Mechanism
+                {
+                    Handled = false,
+                    Type = "unity.log"
+                }
+            };
+        }

```

Comment:
I didn't go through in detail to point out where the opportunities for refactor are but it seems to me there's a lot of duplication here. Particularly in areas that are candidate for change like mechanism.type. Ideally this would be refactored a bit.